cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,228 @@
+"""
+Tests for the maintenance app views.
+"""
+import ddt
+import itertools
+
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext_lazy as _
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from student.roles import GlobalStaff
+from student.tests.factories import AdminFactory, UserFactory
+
+from contentstore.management.commands.utils import get_course_versions
+
+from .views import COURSE_KEY_ERROR_MESSAGES, get_maintenace_urls
+
+
+class CourseKeyMixin(object):
+    def verify_error_message(self, data, error_msg):
+        response = self.client.post(self.view_url, data=data)
+        self.assertContains(response, error_msg, status_code=200)
+
+    def verify_invalid_course_key(self, course_key, error_msg):
+        self.verify_error_message({'course-id': course_key}, error_msg)
+
+    def verify_bad_course_keys(self):
+        self.verify_invalid_course_key('', COURSE_KEY_ERROR_MESSAGES['empty_course_key'])
+        self.verify_invalid_course_key('edx', COURSE_KEY_ERROR_MESSAGES['invalid_course_key'])
+        self.verify_invalid_course_key('course-v1:e+d+X', COURSE_KEY_ERROR_MESSAGES['course_key_not_found'])
+
+    def validate_success_from_response(self, response, success_message):
+        self.assertNotContains(response, '<div class="error">', status_code=200)
+        self.assertContains(response, success_message, status_code=200)
+
+
+class MaintenanceViewTestCase(ModuleStoreTestCase):
+    """
+    Base class for maintenance view tests.
+    """
+
+    def setUp(self):
+        """Create a user and log in. """
+        super(MaintenanceViewTestCase, self).setUp()
+        self.user = AdminFactory()
+        success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(success)
+        self.course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+
+    def tearDown(self):
+        """
+        Reverse the setup
+        """
+        self.client.logout()
+        ModuleStoreTestCase.tearDown(self)
+
+
+@ddt.ddt
+class MaintenanceViewAccessTests(MaintenanceViewTestCase):
+    """
+    Tests for access control of maintenance views.
+    """
+
+    @ddt.data(*(
+        (url_name, role, has_access)
+        for (url_name, (role, has_access))
+        in itertools.product((
```

Comment:
I am checking this for each URL, we have different roles which would either have access or not.